core/state/node/transitions.go

```diff
@@ -109,25 +118,46 @@ func (t *MigrationTransition) Validate(oldState []byte) error {
 }
 
 type AddUserTransition struct {
-	UserID      string `json:"user_id"`
 	Username    string `json:"username"`
 	Certificate string `json:"certificate"`
+
+	EnrichedData *AddUserTranstitionEnrichedData `json:"enriched_data"`
+}
+
+type AddUserTranstitionEnrichedData struct {
+	User *User `json:"user"`
 }
 
 func (t *AddUserTransition) Type() string {
 	return TransitionAddUser
 }
 
 func (t *AddUserTransition) Patch(oldState []byte) ([]byte, error) {
```

Comment:
can we make these function signatures / states into separate types to be more clear about what the return value means?